resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v2/DownloadRemoteDependencyManager.scala

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.kubernetes.submit.v2
+
+import java.io.File
+
+import io.fabric8.kubernetes.api.model.{ConfigMap, ContainerBuilder, EmptyDirVolumeSource, PodBuilder, VolumeMountBuilder}
+
+import org.apache.spark.deploy.kubernetes.config._
+import org.apache.spark.deploy.kubernetes.constants._
+import org.apache.spark.deploy.kubernetes.submit.KubernetesFileUtils
+import org.apache.spark.util.Utils
+
+private[spark] trait DownloadRemoteDependencyManager {
+
+  def buildInitContainerConfigMap(): ConfigMap
+
+  def configurePodToDownloadRemoteDependencies(
+    initContainerConfigMap: ConfigMap,
+    driverContainerName: String,
+    originalPodSpec: PodBuilder): PodBuilder
+
+  /**
+   * Return the local classpath of the driver after all of its dependencies have been downloaded.
+   */
+  def resolveLocalClasspath(): Seq[String]
+}
+
+// TODO this is very similar to SubmittedDependencyManagerImpl. We should consider finding a way to
+// consolidate the implementations.
+private[spark] class DownloadRemoteDependencyManagerImpl(
+    kubernetesAppId: String,
+    sparkJars: Seq[String],
+    sparkFiles: Seq[String],
+    jarsDownloadPath: String,
+    filesDownloadPath: String,
+    initContainerImage: String) extends DownloadRemoteDependencyManager {
+
+  private val jarsToDownload = KubernetesFileUtils.getOnlyRemoteFiles(sparkJars)
+  private val filesToDownload = KubernetesFileUtils.getOnlyRemoteFiles(sparkFiles)
```

Comment:
move these into `buildInitContainerConfigMap` since that's the only place they're used. Otherwise these run in the object constructor which I don't think we want (constructors should be cheap)